pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStages.java

```diff
@@ -38,26 +34,20 @@ public void validate(@Nonnull final ModelValidator validator) {
 
     public void validate(final ModelValidator validator, boolean isWithinParallel) {
         validator.validateElement(this);
-        for (ModelASTStage stage : stages) {
-            stage.validate(validator, isWithinParallel);
+        if (stages != null) {
```

Comment:
Looking at the usage in the class (`equals`) I think `stages` is meant to be `@Nonnull` and the setter should be more restrictive. Either way this doesn't look like a refactoring, maybe handle it separately?